service/k8sclient/k8client.go

```diff
@@ -0,0 +1,330 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package k8sclient
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/AlekSi/pointer"
+	_ "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1" // We will use it later
+	pxc "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/percona-platform/dbaas-controller/logger"
+	kubectl2 "github.com/percona-platform/dbaas-controller/service/k8sclient/kubectl"
+)
+
+const backupImage = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0-backup"
+const pxcImage = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0"
+const backupStorageName = "test-backup-storage"
+
+const perconaXtradbClusterKind = "PerconaXtraDBCluster"
+const perconaServerMongoDBKind = "PerconaServerMongoDB"
+
+// CreateParams contains all parameters required to create cluster.
+type CreateParams struct {
```

Comment:
While PXC and MongoDB clusters share some parameters, not all of them are common. We should not have common types and methods for them.

In https://github.com/percona-platform/dbaas-controller/pull/15#discussion_r455649543 I asked for methods to be renamed to more specific – instead they were made more general.